pages/login/index.js

```diff
@@ -115,10 +158,32 @@ border border-[#50545A] px-4 py-[1.1rem]"
             border border-[#50545A] px-4 py-[1.1rem]"
           >
             <button
-              onClick={discordAuth}
-              className="mx-[1.1rem] h-[5rem] w-[25.2rem] w-full rounded-[1rem] bg-[#5865F2] text-[1.8rem] text-white"
+              onClick={generateDiscordUrl}
+              className={`relative mx-[1.1rem] h-[5rem] w-full rounded-[1rem] bg-[#5865F2] text-[1.8rem] text-black ${
+                gettingDiscordUrl ? 'cursor-wait opacity-50' : ''
+              }`}
             >
-              Login with Discord
+              {gettingDiscordUrl && (
+                <div className="absolute right-0 top-1/2 mr-2 -translate-y-1/2 transform">
+                  <svg
```

Comment:
Please make a reusable component for this since I think it's being used several times across pages thanks! @micohalvarez